### PR TITLE
Feat: added "Fund this package" button

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Type:** Feat

The following has been addressed in the PR: #361

**Description:** Introduced "Fund this package" button. 

image

### CODE REVIEW

**Comment by juanpicado** [comment]
@priscilawebdev it has conflicts ..

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  marginRight: 3,
+});
+
+interface Props {
+  to: string;
+}
+
+/* eslint-disable react/jsx-no-bind */
+const DetailSidebarFundButton: React.FC<Props> = ({ to }) => {
+  const preventDefault = (event: MouseEvent<HTMLButtonElement>) => event.preventDefault();
```
Comment (juanpicado) [inline src/components/DetailSidebar/DetailSidebarFundButton.tsx]: The user clicks the button and then what happens? .... We should open the target in a new tab.

Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: We must also update `@verdaccio/types` to be able to have

```
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/verdaccio"
  },
```
in https://github.com/verdaccio/monorepo/blob/master/core/types/index.d.ts#L132 as optional type. Then can be used here.

Comment (juanpicado) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: You can find all the rules about that field here https://github.com/npm/rfcs/pull/54/files#diff-1bc199539d172d3211c84688a617e18cR30

**Review by juanpicado — COMMENTED** [review]
Looks nice 👍

File: src/components/DetailSidebar/DetailSidebarFundButton.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: @juanpicado thank you for your feedback :) 

The component StyledLink has the prop "external". When this prop is set to true, the anchor element will have target="_blank" .😉

Comment (anikethsaha) [reply src/components/DetailSidebar/DetailSidebarFundButton.tsx]: so is it configurable from the verdaccio config file ?

I would make it default to new tab (\_blank)

**Comment by juanpicado** [comment]
The best would be to have a PR with the whole thing instead spread the feature in multiple PR, this is helpful when we need to track issues or when features were introduced, otherwise the changelog file looks cleaner and anyone can identify the exact version where a feat was introduced.

**Comment by DanielRuf** [comment]
Smaller commits / PRs make bisecting easier (I know, we squash them by default).

**Pull request merged**